### ISSUE TITLE
Type coercion has been deprecated. Please, use data pipes instead.

### ISSUE THREAD

**Issue by rvillalba-novetta** [visjs/vis-timeline#254]
I'm getting the following log printing to my console whenever I update the timeline's DataSet with `dataSet.update()`

```
backend.js:6 Type coercion has been deprecated. Please, use data pipes instead. See https://visjs.github.io/vis-data/data/datapipe.html#TypeCoercion for more details with working migration example.

r | @ | backend.js:6
  | warnTypeCorectionDeprecation | @ | esm.js:14
  | _getItem | @ | esm.js:14
  | get | @ | esm.js:14
  | (anonymous) | @ | ItemSet.js:1146
  | _onUpdate | @ | ItemSet.js:1145
  | update | @ | ItemSet.js:165
  | (anonymous) | @ | esm.js:14
  | _trigger | @ | esm.js:14
  | update | @ | esm.js:14
```

I'm on the latest version of vis-timeline.

**Comment by alebx** [comment]
I also get this warning and I am unable to comprehend what I am doing wrong.
I am simply setting the data via the setData method.

**Comment by anderschbe** [comment]
+1, this is a rather nasty problem, as it added 4-5 ms PER item to our updates. Given that there seems to be no way easy yet to access the DataPipe constructor (at least it's not exported by vis-data, nor easily accessible therefrom), this is blocking.

**Comment by yotamberk** [comment]
@Thomaash Can you please explain how things work with data pipes?

**Comment by Thomaash** [comment]
Hi @rvillalba-novetta, @alebx, @anderschbe and anybody else reading this,

I anticipated this issue and there is a section at the end of the docs (<https://visjs.github.io/vis-data/data/datapipe.html>) to help you with transition. It includes a piece of code that you can copy paste to get very similar functionality. It's not a drop in replacement though because you'll end up with uncoerced (input) and coerced (output) data sets.

Regarding the constructor that is indeed not exported. It is created using a builder that you can obtain by calling `createNewDataPipeFrom` function.

PS: If you still have some questions after reading the docs feel free to mention me.

**Comment by Thomaash** [comment]
Okay so I think I have everything figured out. I didn't realize how heavily dependent Vis Timeline is on this functionality. Once it's reviewed (hopefully there are no more issues in the PRs) everything should work fine.

My previous comment applies only if you use type coersion directly (the `type` map from the Data Set). If you don't it will be resolved by an update soon.

**Comment by Thomaash** [comment]
It is a bug after all.

**Comment by AsafEini** [comment]
Hey, any progress with this bug? it explodes the console with tons for warning and traces.
generally updating the dataSet became very slow.
would be happy to know in which version the coercion issue still hasn't been merged so i can use that version in the meantime..

**Comment by Thomaash** [comment]
6.4.0 introduced this. A fix that would solve this (#259) awaits review.

It's also possible to just update Vis Data as the logging has been reduced a lot in last release.

**Comment by yotamberk** [comment]
Fixed with #259
